### ISSUE TITLE
Add footnote when shapiro-wilk doesn't work when n > 5000

### ISSUE THREAD

**Issue by IndrajeetPatil** [jamovi/jmv#160]
Shapiro-Wilk test doesn't seem to work when there are more than 5000 data points. 

``` r
library(jmv)
library(nortest)

dat <- data.frame(x = c(rep(1, 2500), rep(2, 2501)), y = rnorm(5001))

jmv::ttestOneS(data = dat,
               vars = "y",
               norm = TRUE)
#> 
#>  ONE SAMPLE T-TEST
#> 
#>  One Sample T-Test                                  
#>  -------------------------------------------------- 
#>                        statistic    df      p       
#>  -------------------------------------------------- 
#>    y    Student's t        -1.15    5000    0.251   
#>  -------------------------------------------------- 
#> 
#> 
#>  Test of Normality (Shapiro-Wilk) 
#>  -------------------------------- 
#>         W      p       
#>  -------------------------------- 
#>    y    NaN            
#>  -------------------------------- 
#>    Note. A low p-value
#>    suggests a violation of
#>    the assumption of
#>    normality

stats::shapiro.test(x = dat$y)
#> Error in stats::shapiro.test(x = dat$y): sample size must be between 3 and 5000

nortest::ad.test(dat$y)
#> 
#>  Anderson-Darling normality test
#> 
#> data:  dat$y
#> A = 0.30397, p-value = 0.5714
```

Created on 2018-03-18 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by jonathon-love** [comment]
yeah, that's a limitation built into R. not much we can do about it.

**Comment by IndrajeetPatil** [comment]
If you don't mind having another package dependency, `nortest::ad.test` can bail you out. Or you can just implement your own version of that test. It won't fail with _n_ > 5000.

**Comment by jonathon-love** [comment]
yup. i did look into this a while ago (like, possibly 4 years ago). i think the issue with shapiro-wilk is that it's computed using an approximation, and that approximation isn't good, when n > 5000. so it's not simply a matter of removing that upper bound.

but i'm kinda down on assumption tests anyway. this result is non-significant, so we conclude that the data isn't definitely not normal... and how do you adjust for multiple comparisons!

i think the APA discourage people from normality tests too, and favour the q-q plot.

what do you think? do we really need a normality test for n > 5000?

**Comment by IndrajeetPatil** [comment]
I personally don't think we need normality tests with n > 5000 for sure! I am in general highly skeptical of normality tests, I just wanted to let you know of this issue in case you hadn't come across it before as we rarely have datasets in psychology with more than 5000 datapoints. 

I think it will be good if you actually display some informative error if _n_ > 5000 so that the user knows that there is nothing wrong with their data, but the test itself is no longer valid. 

P.S. I raised this issue here because it was raised for my package (https://github.com/IndrajeetPatil/ggstatsplot/issues/24) and most of my statistical tests in there are dependent on `jmv` so wanted to first check it here. I resolved it by adding another dependency `nortest`, which has the AD test, but given my love for `jamovi` I wanted to inform you of this as well :)
